examples/nlp/sentence_embeddings_with_sbert.py

```diff
@@ -0,0 +1,472 @@
+"""
+Title: Sentence embeddings using Siamese BERT-networks
+Author: [Mohammed Abu El-Nasr](https://github.com/abuelnasr0)
+Date created: 2023/06/20
+Last modified: 2023/06/20
+Description: Fine-tune a RoBERTa model to generate sentence embeddings using KerasNLP.
+"""
+"""
+## Introduction
+
+BERT and RoBERTa can be used for semantic textual similarity tasks, where two sentences
+are passed to the model and the network predicts whether they are similar or not. But
+what if we have a collection of a large number of sentences and we want to find the most
+similar pairs in that collection? That will take n*(n-1)/2 inference computations, where
+n is the number of sentences in the collection. For example, if n = 10000, the required
+time will be 65 hours on a V100 GPU.
+
+A common method to overcome the time overhead issue is to pass one sentence to the model,
+then average the output of the model, or take the first token (the [CLS] token) and use
+them as [sentence embedding](https://en.wikipedia.org/wiki/Sentence_embedding), then use
+a vector similarity measure like cosine similarity or Manhatten / Euclidean distance to
+find close sentences (semantically similar sentences). That will reduce the time to find
+the most similar pairs in a collection of 10,000 sentences from 65 hours to 5 seconds!
+
+If we use BERT or RoBERa directly, that will yield rather bad sentence embeddings. But if
```

Comment:
RoBERa -> RoBERTa

Also, it's kind hard to track if this is example is going to be about BERT or RoBERTa. Maybe replace all reference to BERT -> RoBERTa in this paragraph.